src/Microsoft.DotNet.Tools.Compiler.Native/ILCompilerInvoker.cs

```diff
@@ -54,7 +46,8 @@ private void InitializeArgs(NativeCompileSettings config)
             argsList.Add($"\"{inputFilePath}\"");
             
             // System.Private.CoreLib Reference
-            var coreLibPath = Path.Combine(config.IlcPath, OSCoreLibNameMap[config.OS]);
+            String[] coreLibs = new String[] { "System.Private.CoreLib.dll", "System.Private.Corelib.dll" };
+            var coreLibPath = Path.Combine(config.IlcPath, Array.Find(coreLibs, lib => File.Exists(Path.Combine(config.IlcPath, lib))));
```

Comment:
Hmmm I guess as long as the minor version is bumped on the ILCompiler package each time this works. Nice :smile: